ext/vulnsrc/alpine/alpine.go

```diff
@@ -195,42 +132,59 @@ type secDBFile struct {
 	} `yaml:"packages"`
 }
 
-func parseYAML(r io.Reader) (vulns []database.VulnerabilityWithAffected, err error) {
-	var rBytes []byte
-	rBytes, err = ioutil.ReadAll(r)
+func parseSecDBFile(filePath string) (file *secDBFile, err error) {
+	var f io.ReadCloser
+	f, err = os.Open(filePath)
 	if err != nil {
 		return
 	}
 
-	var file secDBFile
-	err = yaml.Unmarshal(rBytes, &file)
+	defer f.Close()
+	data, err := ioutil.ReadAll(f)
 	if err != nil {
 		return
 	}
 
-	for _, pack := range file.Packages {
-		pkg := pack.Pkg
-		for version, vulnStrs := range pkg.Fixes {
-			err := versionfmt.Valid(dpkg.ParserName, version)
-			if err != nil {
-				log.WithError(err).WithField("version", version).Warning("could not parse package version. skipping")
+	file = &secDBFile{}
+	err = yaml.Unmarshal(data, file)
+	return
+}
```

Comment:
Prefer decoders when available to Unmarshal.
Decoders stream data and can avoid ballooning memory when processing large files.
I know these aren't very large right now, but they could become large at some point in the future.

```go
func parseSecDBFile(filePath string) (file *secDBFile, err error) {
  var f io.ReadCloser
  f, err = os.Open(filePath)
  if err != nil {
    return
  }
  defer f.Close()

  file = &secDBFile{}
  err = yaml.NewDecoder(f).Decode(&file)
  return
}
```